### DISCUSSION TITLE
How do I use inline math as the contents in a diagram box using mermaid?

### DISCUSSION THREAD

**Discussion by tony-an-phd** [quarto-dev/quarto-cli#11395] [Q&A] [answered]
### Description

I want to display inline math inside a box of a diagram using mermaid, but it won't render 
```{mermaid}
flowchart LR
  A[$w_{ijk}$] --> B[$p_{ijk}^{1}$]
  B --> C[new contents]
```
Any trick I can get around it? I tried removing the $ and it didn't work either. It is displayed as 
![image](https://github.com/user-attachments/assets/fb094ec5-f35c-4cc2-a553-3c10a57f8efb)

**Comment by mcanouil** [comment, marked as answer]
Could you provide a complete small reproducible example rather than a standalone snippet of code? Also what version of Quarto are you using?

Also, see previous discussion:
- https://github.com/quarto-dev/quarto-cli/discussions/10851

Note that your diagram is not valid (independently of Quarto).
You need to write valid Mermaid.js diagram, see Mermaid.js documentation.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by mcanouil** [reply]
Thanks for the additional information.

Again:

> Note that your diagram is not valid (independently of Quarto).
You need to write valid Mermaid.js diagram, see Mermaid.js documentation.

https://mermaid.js.org/config/math.html

edit: this is valid syntax:

```
flowchart LR
  A["$$w_{ijk}$$"] --> B["$$p_{ijk}^{1}$$"]
  B --> C[new contents]
```
